File: tests/e2e/test_frontend.py
"""Frontend E2E tests using Playwright."""
import asyncio
from playwright.async_api import async_playwright


async def test_frontend_screenshot():
    """Take a screenshot of the frontend to verify it's working."""
    async with async_playwright() as p:
        browser = await p.chromium.launch()
        page = await browser.new_page()
        
        # Navigate to frontend
        await page.goto("http://localhost:3000")
        
        # Wait for the main heading to be visible
        await page.wait_for_selector("h1:has-text('Academic Citation Assistant')")
        
        # Take screenshot
        await page.screenshot(path="tmp/frontend-screenshot.png")
        
        # Check for key elements
        assert await page.is_visible("text=Real-time Citation Recommendations")
        assert await page.is_visible("text=Real-time Suggestions")
        assert await page.is_visible("text=Context-Aware")
        assert await page.is_visible("text=Confidence Scoring")
        
        # Check console for errors
        console_messages = []
        page.on("console", lambda msg: console_messages.append(msg))
        
        await browser.close()
        
        print("✅ Frontend is working correctly!")
        print(f"Screenshot saved to tmp/frontend-screenshot.png")
        
        # Report any console errors
        errors = [msg for msg in console_messages if msg.type == "error"]
        if errors:
            print(f"⚠️  Found {len(errors)} console errors:")
            for error in errors:
                print(f"   - {error.text}")
        else:
            print("✅ No console errors found")


if __name__ == "__main__":
    asyncio.run(test_frontend_screenshot())

File: frontend/test-frontend.mjs
import { chromium } from '@playwright/test';

async function testFrontend() {
  const browser = await chromium.launch({ headless: false });
  const context = await browser.newContext();
  const page = await context.newPage();
  
  try {
    console.log('1. Navigating to frontend...');
    await page.goto('http://localhost:3002');
    
    // Take screenshot of homepage
    await page.screenshot({ path: 'test-homepage.png' });
    console.log('✓ Homepage loaded');
    
    // Check if page loads without errors
    await page.waitForTimeout(2000);
    
    // Check console for errors
    page.on('console', msg => {
      if (msg.type() === 'error') {
        console.error('Console error:', msg.text());
      }
    });
    
    // Try to navigate to documents
    console.log('2. Navigating to documents...');
    await page.click('text=My Documents');
    await page.waitForTimeout(1000);
    await page.screenshot({ path: 'test-documents.png' });
    console.log('✓ Documents page loaded');
    
    // Try to navigate to library
    console.log('3. Navigating to paper library...');
    await page.click('text=Paper Library');
    await page.waitForTimeout(1000);
    await page.screenshot({ path: 'test-library.png' });
    console.log('✓ Paper library loaded');
    
    // Check that Zotero Sync button is removed
    const zoteroButton = await page.locator('text=Zotero Sync').count();
    if (zoteroButton === 0) {
      console.log('✓ Zotero Sync button successfully removed');
    } else {
      console.error('✗ Zotero Sync button still exists!');
    }
    
    console.log('\nTest complete! Check the screenshots.');
    
  } catch (error) {
    console.error('Test failed:', error);
    await page.screenshot({ path: 'test-error.png' });
  } finally {
    await browser.close();
  }
}

testFrontend();